src/main/java/com/avispl/symphony/dal/avdevices/mixer/vaddio/easyip/EasyIPMixerCommunicator.java

```diff
@@ -0,0 +1,1436 @@
+		this.setCommandSuccessList(Collections.singletonList("> "));
+		this.setLoginSuccessList(Collections.singletonList("> "));
+		this.setLoginErrorList(Collections.singletonList("Permission denied, please try again."));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+		List<AdvancedControllableProperty> advancedControllableProperties = new ArrayList<>();
+		Map<String, String> stats = new HashMap<>();
+		Map<String, String> controlStats = new HashMap<>();
+		reentrantLock.lock();
+		try {
+			if (!isEmergencyDelivery) {
+				convertConfigManagement();
+				convertEnableCrosspointGain();
+				retrieveAllData();
+				populateAllData(stats, controlStats, advancedControllableProperties);
+				if (isConfigManagement) {
+					stats.putAll(controlStats);
+					extendedStatistics.setControllableProperties(advancedControllableProperties);
+				}
+				extendedStatistics.setStatistics(stats);
+				localExtendedStatistics = extendedStatistics;
+			}
+			isEmergencyDelivery = false;
+		} finally {
+			reentrantLock.unlock();
+		}
+		return Collections.singletonList(localExtendedStatistics);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+		String property = controllableProperty.getProperty();
+		String value = String.valueOf(controllableProperty.getValue());
+		Map<String, String> stats = this.localExtendedStatistics.getStatistics();
+		List<AdvancedControllableProperty> advancedControllableProperties = this.localExtendedStatistics.getControllableProperties();
+		if (logger.isDebugEnabled()) {
+			logger.debug(String.format("Perform control operation with property: %s and value: %s", property, value));
+		}
+		reentrantLock.lock();
+		try {
+			if (localExtendedStatistics == null) {
+				if (logger.isDebugEnabled()) {
+					logger.debug(String.format("Error while controlling %s metric", property));
+				}
+				return;
+			}
+			isEmergencyDelivery = true;
+			String group = EasyIPMixerConstant.EMPTY;
+			String[] propertyList = property.split(EasyIPMixerConstant.HASH);
+			String propertyKey = property;
+			if (property.contains(EasyIPMixerConstant.HASH)) {
+				propertyKey = propertyList[1];
+				group = propertyList[0];
+			}
+
+			if (group.contains(EasyIPMixerConstant.CROSSPOINT)) {
+				//Crosspoint
+				String command;
+				String inputPropertyName;
+				String outputPropertyName = group.replace(EasyIPMixerConstant.CROSSPOINT, EasyIPMixerConstant.EMPTY);
+				if (propertyKey.contains(EasyIPMixerConstant.GAIN)) {
+					inputPropertyName = propertyKey.replace(EasyIPMixerConstant.GAIN_DB, EasyIPMixerConstant.EMPTY);
+					command = EasyIpMixerCommand.GAIN_CONTROL.replace("$1", AudioOutput.getValueByName(outputPropertyName)).replace("$2", AudioInput.getValueByName(inputPropertyName)).replace("$3", value);
```

Comment:
> For string templates in general, you could've used String.format("", a, b, c)
> 
> It's fine to leave it as it is, but it may be a more convenient alternative in the future :) (you won't need to keep in mind, whether it's a numbered one, and which one specifically)

Sure, I will use it in the future. Thank you